### PR TITLE
Clarify difference between `QuantumTape` and `QuantumScript`

### PR DESCRIPTION
**Pull request by albi3ro**
[sc-54022]

**Context:**

We keep the name `QuantumTape` in documentation, but strictly use its parent `QuantumScript` in PennyLane source code. We use `QuantumScript` because carrying around an unnecessary `AnnotatedQueue` can cause difficulties with parallelism and construction via queuing is substantially more expensive in both memory and time. Queuing can also lead to unintended side effects of things entering the circuit that shouldn't, or things not entering the circuit that should.

This disconnect between documentation and source code is not explained anyone, and a bit confusing.  So this adds some clarification to the module documentation for `qml.tape.`.

**Description of the Change:**

Added module documentation.

**Benefits:**

Hopefully this clears up some confusion.

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/code/qml_tape.rst [hunk 1]
```diff
@@ -18,6 +18,51 @@ TensorFlow, and PyTorch.
     details on creating QNodes, as well as the :func:`~pennylane.qnode` decorator
     and :func:`~pennylane.QNode` constructor.
 
+**``QuantumTape`` versus ``QuantumScript``**
```
Comment (trbromley) [inline doc/code/qml_tape.rst:21]:
```suggestion
QuantumTape versus QuantumScript
--------------------------------
```

File: doc/code/qml_tape.rst [hunk 2]
```diff
@@ -18,6 +18,51 @@ TensorFlow, and PyTorch.
...
+:class:`~.QueuingManager`, the ``QuantumTape`` requires a ``threading.RLock`` which complicates its use in distributed
+situations.
+
+>>> with QuantumTape(shots=10) as tape:
+...     qml.PauliX(0)
+...     qml.expval(qml.PauliZ(0))
+>>> tape
+<QuantumTape: wires=[0], params=0>
+
+The ``QuantumTape`` also carries around the unprocessed queue in addtion to the processed ``operations`` and ``measurements``, leading
```
Comment (trbromley) [inline doc/code/qml_tape.rst:42]:
```suggestion
The ``QuantumTape`` also carries around the unprocessed queue in addition to the processed ``operations`` and ``measurements``, 
```

**Review by trbromley — APPROVED** [review]
Thanks @albi3ro!

File: doc/code/qml_tape.rst [hunk 3]
```diff
@@ -18,6 +18,51 @@ TensorFlow, and PyTorch.
...
+
+A ``QuantumScript`` is purely a representation of a quantum circuit, and can only be constructed
+via initialization. Once it is initialized, the contents should then remain immutable throughout its lifetime.
+
+>>> ops = [qml.PauliX(0)]
+>>> measurements = [qml.expval(qml.PauliZ(0))]
+>>> QuantumScript(ops, measurements, shots=10)
+<QuantumScript: wires=[0], params=0>
+
+A ``QuantumTape`` has additional queuing capabilities and also inherits from :class:`~.AnnotatedQueue`.  It's contents
```
Comment (albi3ro) [inline doc/code/qml_tape.rst:31]:
```suggestion
A ``QuantumTape`` has additional queuing capabilities and also inherits from :class:`~pennylane.queuing.AnnotatedQueue`.  It's contents
```

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -75,6 +75,8 @@
 
 <h3>Documentation 📝</h3>
 
+* The module documentation for `pennylane.tape` now explains the difference between `QuantumTape` and `QuantumScript`.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:78]:
```suggestion
* The module documentation for `pennylane.tape` now explains the difference between `QuantumTape` and `QuantumScript`.
  [(#5065)](https://github.com/PennyLaneAI/pennylane/pull/5065)
```

**Pull request merged**